certification/internal/policy/operator/certified_images.go

```diff
@@ -0,0 +1,98 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/operator-framework/operator-manifest-tools/pkg/image"
+	"github.com/operator-framework/operator-manifest-tools/pkg/pullspec"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/pyxis"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type imageFinder interface {
+	FindImagesByDigest(context.Context, []string) ([]pyxis.CertImage, error)
+}
+
+// CertifiedImagesCheck scans the CSV and validates that all refereenced images are certified.
+type certifiedImagesCheck struct {
+	ImageFinder        imageFinder
+	nonCertifiedImages []string
+}
+
+func NewCertifiedImagesCheck(imageFinder imageFinder) *certifiedImagesCheck {
+	return &certifiedImagesCheck{
+		ImageFinder: imageFinder,
+	}
+}
```

Comment:
My need to shuffle the doc here a bit? Documentation on the `New...` func should get picked up by godoc... but also this is an `internal` package so I'm not sure if that's true.

Reply:
Done.